### PR TITLE
[CUDA][CodeGen] Fix cuda codegen's fp16 inf literal

### PR DESCRIPTION
**Pull request by wrongtest-intellif**
Try fix https://github.com/apache/tvm/issues/12566

### CODE REVIEW

**Review by vinx13 — APPROVED** [review]

**Comment by vinx13** [comment]
cc @shingjan

**Review by shingjan — APPROVED** [review]
LGTM. Thanks for sending this fix!

File: src/target/source/codegen_cuda.cc [hunk 1]
```diff
@@ -1197,8 +1197,10 @@ inline void PrintConst(const FloatImmNode* op, std::ostream& os, CodeGenCUDA* p)
       break;
     }
     case 16: {
-      os << "__float2half_rn";
-      os << '(' << std::scientific << op->value << 'f' << ')';
+      os << "__float2half_rn" << '(';
```
Comment (shingjan) [inline src/target/source/codegen_cuda.cc:1200]: Would that be possible that we can add some test? We can probably re-use my repro [here](https://gist.github.com/shingjan/9ee06597a13f9be324828afb6c4560b8)

Comment (wrongtest-intellif) [reply src/target/source/codegen_cuda.cc:1200]: the case is added

**Pull request merged**